### DISCUSSION TITLE
Some Questions...

### DISCUSSION THREAD

**Discussion by Kimet1970** [djyt/cannonball#98] [General]
I have 2 personal Cannonball packages in my hard disk...
The old version (32 & 64 bits) v.0.3 (I have an old 32 bits computer) and the last 0.33 version (only 64 bits)...

In my personal packages (a zip file) I keep
- "Microsoft DirectX 8.1 SDK Full (Force Feedback)"
- "Microsoft Visual C++ 2015-2019 Redistributable"...

But actually I never needed to install these packages!
**Actually, I needed to put four extra dlls in my Cannonball folder: Vcruntime140_1, Vcruntime140, Msvcp140_1, Msvcp140**
**Since I added theses dlls Cannonball works fine without necessity to install C++ or Direct-X...**

**My questions are:**
- The C++ package is really necessary? (Or we have only to add the extra dlls?)
- The last version of Cannonball really needs some piece direct-X code? And the old version?

Can I to delete it from my packages?

**Comment by djyt** [comment]
- You definitely don't need the DirectX SDK 8.1, unless you are planning on compiling the code yourself.
- You need the Visual C++ 2015-2019 Redistributable installed (x64 version).

**Reply by Kimet1970** [reply]
Ok Thanks!.
Thus, I think pieces of Visual C++ Redistributables code are already pre-installed in the Windows 10 (or through Windows 10 updates)...
Because I never installed it in my Windows 10...
And it's a recently reformatted system... with a very very  "clean" Windows 10 installation - only very few portable programs (very possibly I didn't install C++ Redistributables by installing other softwares)...
And I could to play Cannonbal only getting the requested missing dll files!
